engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/IsSameObjectNode.java

```diff
@@ -46,6 +52,16 @@ boolean isSameMetaObjects(
   @Fallback
   boolean isIdenticalObjects(
       Object left, Object right, @CachedLibrary(limit = "2") InteropLibrary interop) {
-    return (left == right) || interop.isIdentical(left, right, interop);
+    if (left == right) {
+      return true;
+    }
+    if (interop.isString(left) && interop.isString(right)) {
+      try {
+        return interop.asString(left).equals(interop.asString(right));
+      } catch (UnsupportedMessageException ex) {
+        // go on
```

Comment:
That doesn't seem like a normal behaviour so probably should propagate `ex` via `IllegalStateException`, right?

Reply:
That would change the semantics. The code continues with `isIdentity` check as before.

Reply:
So it is the `equals` which may throw `UnsupportedMessageException`? Because we seem to check that `left` and `right` are strings a few lines before.
I just don't think that exceptions being part of a regular control flow is good in general and I would preferred if this was guarded somehow explicitly.